### ISSUE TITLE
Forms always render the same element

### ISSUE THREAD

**Issue by pcdavid** [eclipse-sirius/sirius-web#3061]
In `FormEventProcessor.refreshForm()`, we currently do:
```java
VariableManager variableManager = new VariableManager();
variableManager.put(VariableManager.SELF, this.formCreationParameters.getObject());
```
In other words, the `self` element on which the `FormRenderer` will be invoked will always be the initial object passed when the form was initialy created.

This works "by mistake" in the standard case where this `self` object is a mutable `EObject`: each render sees the same object, but each time in the correct/expected state.

However this breaks when the target object is immutable, and a new, different Java object is used to represent the successive states of that object (with the same id). For example when [creating a Form on the content of a Portal representation](#2890). In such a case, the `FormEventProcessor` will always render the same `Form` on every refresh as the original object from `formCreationParameters` is immutable.

Diagrams solve this (not sure that was exactly on purpose) by re-resolving the target object from its id on every refresh:
```java
var optionalObject = this.objectService.getObject(editingContext, previousDiagram.getTargetObjectId());
```

Forms (and other representations which may have the issue) should do the same.

**Comment by pcdavid** [comment]
From a quick look at the other RepresentationEventProcessors, it looks like they are all fine (they probably took DiagramEventProcessor as a base/inspiration), *except* for `SelectionEventProcessor`, which resolves its `selectionConfiguration.getTargetObjectId()` only once in `SelectionEventProcessorFactory.createRepresentationEventProcessor(Class<T>, IRepresentationConfiguration, IEditingContext)` and then reuses the same Java `Object` for all refreshes in `SelectionEventProcessor`.
I'll open a separate issue but this is much more minor.
